integrationtest/backend/tck/src/main/java/org/hibernate/search/integrationtest/backend/tck/testsupport/types/NormalizedStringFieldTypeDescriptor.java

```diff
@@ -71,13 +71,10 @@
 	public Optional<FieldSortExpectations<String>> getFieldSortExpectations() {
 		boolean normalize = TckConfiguration.get().getBackendFeatures().normalizeStringMissingValues();
 
-		return Optional.of( new FieldSortExpectations<>(
-				"Cecilia", "george", "Stefany",
-				"aaron",
-				( normalize ) ? "Daniel" : "daniel",
-				( normalize ) ? "Roger" : "roger",
-				"zach"
-		) );
+		return Optional.of( ( normalize ) ?
+				new FieldSortExpectations<>( "Cecilia", "george", "Stefany", "aaron", "Daniel", "Roger", "zach" ) :
+				new FieldSortExpectations<>( "cecilia", "george", "stefany", "aaron", "daniel", "roger", "zach" )
```

Comment:
This looks wrong. Originally the first three values were about testing sorts, whithout considering missing values.
Then you added something that uses them as replacement for missing values, and when it failed, you lowercased everything.

As a result, we no longer test sorts with varying case in Elasticsearch, even when missing values are not involved and it's supposed to work. That's not a great idea.

Maybe you could move the lowercasing operation to the test method that needs it?
I.e. do something like:

```java
docRefHits = simpleQuery( b -> b.byField( fieldPath ).asc().onMissingValue().use( normalizeIfNecessary( fieldModel.document2Value.indexedValue ) ) ).fetchHits(); 
```

and add this method:

```java
@SuppressWarning("unchecked")
public <T> normalizeMissingValueReplacementIfNecessary(T value) {
    if ( !TckConfiguration.get().getBackendFeatures().normalizeStringMissingValues() && String.class.isInstance ( value ) ) {
        // The backend doesn't normalize missing value replacements automatically, we have to do it ourselves
        // TODO HSEARCH-3387 Remove this once all backends correctly normalize missing value replacements
        return (T) missingValue.toLowercase( Locale.ROOT );
    }
    else {
        return value;
    }
}
```

Reply:
You're right! Predicate values are not normalized, but  indexed values are.